### DISCUSSION TITLE
Negative PnL Strategy's Sharpe, Sortino and Calmar ratios are zero

### DISCUSSION THREAD

**Discussion by atakanokan** [kernc/backtesting.py#472] [Q&A] [answered]
Is there any way for Backtest.run() to return values (instead of 0) for strategies with negative PnL? From what I can see, they all return 0.

**Comment by kernc** [comment, marked as answer]
You could remove `np.clip()` wrapping in these places:
https://github.com/kernc/backtesting.py/blob/7671f044d554249b608b3241dd7cca37ec5a35d1/backtesting/_stats.py#L120
https://github.com/kernc/backtesting.py/blob/7671f044d554249b608b3241dd7cca37ec5a35d1/backtesting/_stats.py#L122
https://github.com/kernc/backtesting.py/blob/7671f044d554249b608b3241dd7cca37ec5a35d1/backtesting/_stats.py#L124
Do [negative Sharpe, Sortino, Calmar ratios make any sense](https://www.google.com/search?q=negative+(sharpe+OR+sortino+OR+calmar)+ratio)?

**Reply by atakanokan** [reply]
It would be good to look at cross-asset robustness via SR distribution which Neg->0 SR clipping skews. Would be great to receive the ratios as is and do clipping if needed.

**Reply by atakanokan** [reply]
@kernc if you think this would be a good PR, I would be more than happy to do it

**Reply by kernc** [reply]
Clipping came in in https://github.com/kernc/backtesting.py/commit/028f02d4c5c358e39d67411eeb1f1a5f82370cef with no elaborate justification. [I agree](https://www.google.com/search?q=sharpe+ratio+distribution&tbm=isch), PR welcome. :+1:

**Reply by kernc** [reply]
I remember my thinking: with risk-free rate default assumed 0%, if your strategy's Sharpe is negative, it's better to just divest and keep the money. :smile:

**Reply by atakanokan** [reply]
Opened: https://github.com/kernc/backtesting.py/pull/473
